subsys/net/ip/net_shell.c

```diff
@@ -3532,6 +3532,180 @@ static int cmd_net_ipv6(const struct shell *shell, size_t argc, char *argv[])
+
+	if (net_addr_pton(AF_INET6, argv[2], &addr)) {
+		PR_ERROR("Invalid address: %s\n", argv[2]);
+		return -EINVAL;
+	}
+
+	if (!net_if_ipv6_addr_add(iface, &addr, NET_ADDR_MANUAL, 0)) {
+		PR_ERROR("Failed to add %s address to interface %p\n", argv[2], iface);
+	}
+
+#else /* CONFIG_NET_NATIVE_IPV6 */
+	PR_INFO("Set %s and %s to enable native IPv6 support.\n",
+	 "CONFIG_NET_NATIVE", "CONFIG_NET_IPV6");
+#endif /* CONFIG_NET_NATIVE_IPV6 */
+	return 0;
+}
+
+static int cmd_net_ip6_del(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if *iface = NULL;
+	int idx;
+	struct in6_addr addr;
+
+	if (argc < 3) {
+		PR_ERROR("Correct usage: net ipv6 del <index> <address>\n");
+		return -EINVAL;
+	}
+
+	idx = get_iface_idx(shell, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	if (net_addr_pton(AF_INET6, argv[2], &addr)) {
+		PR_ERROR("Invalid address: %s\n", argv[2]);
+		return -EINVAL;
+	}
+
+	if (!net_if_ipv6_addr_rm(iface, &addr)) {
+		PR_ERROR("Failed to delete %s\n", argv[2]);
+		return -1;
+	}
+
+#else /* CONFIG_NET_NATIVE_IPV6 */
+	PR_INFO("Set %s and %s to enable native IPv6 support.\n",
+	 "CONFIG_NET_NATIVE", "CONFIG_NET_IPV6");
+#endif /* CONFIG_NET_NATIVE_IPV6 */
+	return 0;
+}
+
+static int cmd_net_ip6_route_add(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if *iface = NULL;
+	int idx;
+	int prefix_len = 0;
+	struct net_route_entry *route;
+	struct in6_addr gw = {0};
+	struct in6_addr prefix = {0};
+
+	if (argc < 4) {
+		PR_ERROR("Correct usage: net route add <index> "
+		"<destination> <gateway>\n");
+		return -EINVAL;
+	}
+
+	idx = get_iface_idx(shell, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	if (net_addr_pton(AF_INET6, argv[2], &prefix)) {
+		PR_ERROR("Invalid address: %s\n", argv[2]);
+		return -EINVAL;
+	}
+
+	if (net_addr_pton(AF_INET6, argv[3], &gw)) {
+		PR_ERROR("Invalid gateway: %s\n", argv[3]);
+		return -EINVAL;
+	}
+
+	route = net_route_add(iface,
+					&prefix,
+					NET_IPV6_DEFAULT_PREFIX_LEN,
+					&gw,
+					NET_IPV6_ND_INFINITE_LIFETIME,
+					NET_ROUTE_PREFERENCE_MEDIUM);
+	if (route == NULL) {
+		PR_ERROR("Failed to add route\n");
+		return -ENOEXEC;
+	}
+
+#else /* CONFIG_NET_NATIVE_IPV6 */
+	PR_INFO("Set %s and %s to enable native IPv6 support.\n",
+	 "CONFIG_NET_NATIVE", "CONFIG_NET_IPV6");
+#endif /* CONFIG_NET_NATIVE_IPV6 */
+	return 0;
+}
+
+static int cmd_net_ip6_route_del(const struct shell *shell, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if *iface = NULL;
+	int idx;
+	int prefix_len = 0;
+	struct net_route_entry *route;
+	struct in6_addr prefix = { 0 };
+
+	if (argc < 3) {
```

Comment:
I'd just check for the exact number of expected arguments `(argc != 3)`, same in other commands.

Reply:
Done.